paymentsheet/src/main/java/com/stripe/android/link/ui/signup/SignUpViewModel.kt

```diff
@@ -278,7 +294,8 @@ internal class SignUpViewModel @Inject constructor(
         fun factory(
             parentComponent: NativeLinkComponent,
             navigateAndClearStack: (LinkScreen) -> Unit,
-            moveToWeb: () -> Unit
+            moveToWeb: () -> Unit,
+            dismissWithResult: (LinkActivityResult) -> Unit
```

Comment:
It seems to me that passing in a lambda to a VM factory is an anti-pattern since it can easily leak the Activity (lambda captures an Activity; ViewModel is reused on configuration change). Fortunately, the lambdas we're passing always happen to capture only other Activity ViewModels, but I think it's a footgun we should disarm.

Reply:
100% agreed. We recently refactored navigation away from this pattern in Link in favor of `NavigationManager` (flow-based) - I'll move all existing lambdas (unfortunately that's the current pattern) to a similar approach in a separate PR if that works!